storage/blob.go

```diff
@@ -649,7 +649,14 @@ func (b BlobStorageClient) CreateBlockBlobFromReader(container, name string, siz
 //
 // See https://msdn.microsoft.com/en-us/library/azure/dd135726.aspx
 func (b BlobStorageClient) PutBlock(container, name, blockID string, chunk []byte) error {
-	return b.PutBlockWithLength(container, name, blockID, uint64(len(chunk)), bytes.NewReader(chunk))
+	return b.PutBlockWithLength(container, name, blockID, uint64(len(chunk)), bytes.NewReader(chunk), nil)
+}
+
+func (b BlobStorageClient) PutBlockWithContentType(container, name, blockID string, chunk []byte, contentType string) error {
```

Comment:
Agree with the having only the method below. 

It is also probably a good idea to merge the extra headers into the default, overriding if key matches.

Thoughts?